rest/docs/design/technical_design.md

```diff
@@ -0,0 +1,127 @@
+
+From a UX point of view, the user is exposed to a graphical UI, dynamically built base on the above models.
+Interacting with a set of forms and UI elements, a user completes the definition of a logical integration.
+
+When she's done, the UI application produces a `.json` files that corresponds to the user activity and that will be used as the input to generate code on his behalf.
+
+The component responsible of consuming this `.json` is the [`ActivateHandler`](https://github.com/syndesisio/syndesis/blob/master/rest/controllers/src/main/java/io/syndesis/controllers/integration/online/ActivateHandler.java), that at runtime, lives in the `syndesis-rest` pod.  
+
+This component is responsible of 3 main activities:  
+1. produce **deplyoment data** 
+1. **build** the project related to our `Integration`
+1. **deploy** an instance of our `Integration`
+
+
+
+##### Build
+**Building** is actually delegated to an implementation of [`ProjectGenerator`](https://github.com/syndesisio/syndesis/blob/master/rest/project-generator/src/main/java/io/syndesis/project/converter/DefaultProjectGenerator.java) that is responsible of **build time code generation**.  
+With some introspection and templating, a full Java based Apache Maven project is generated. The idea is that this artifact could be built succesfully even outside Syndesis runtime platform, being a fully consistent project.  
+The `ActivateHandler`, uses the physical bytes generated by the `ProjectGenerator` to trigger a **build** invocation at **Openshift** level.  
+This means that we leverage **Openshift** "source to image (s2i)" capability, that accepts a reference to a source code project location, to build the project at source code level and produce also a corresponding **Docker** image including everything needed to spawn a runtime instance of our `Integration`.
+
+##### Deploy
+**Deploying**, as another responsibility of `ActivateHandler`. After a source code project corresponding to an `Integration` has been generated, built and shared to the interal repository, it still needs to be deployed to run and be available.
+This phase does just that: instructs **Openshift** platform to deploy a live incarnation of our logical `Integration`
+
+
+##### Deployment Data
+This phase actually occur before the `build` one, but it's better understood after what happens in the building phase has been made clearer.  
+We have seen that the building phase is all about code generation, to produce an output that will then pass further to a step of code compilation.  
+Before the code generation phase, actually some further manipulation happens.  
+The idea is to extract externalize from the code itself references to sensible information, like passwords, and externalized properties.  
```

Comment:
"The idea is to extract externalize"
I think you need to choose between these 2 verbs and keep a single one. I think extract is better suited.